### PR TITLE
fix: Ignore hashbang URLs for skiplinks

### LINKED ISSUE #824: False positive: Angular hasbang links are being treated as in-page anchor links
The new aXe plugin for Chrome displays a violation that says “the skip-link target should exist and be focusable” for any Angular links with a hashbang. I am proposing that Angular hashbang links should be excluded. 

The false positive can be tested on the Angular bootstrap page: 
https://ng-bootstrap.github.io/#/getting-started

**Element source**
<a class="navbar-brand" href="#/">ng-bootstrap</a>

**Issue description**
Ensure all skip links have a focusable target

Note: this issue occurs within HTML_CodeSniffer as well and it is one of the main reasons our team  moved away from CodeSniffer and started using aXe.

### PR DESCRIPTION
**Pull request by WilcoFiers**
Closes #824

### CODE REVIEW

**Comment by marcysutton** [comment]
Looks like there are some failing tests we'll need to resolve.

File: test/checks/navigation/internal-link-present.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ describe('internal-link-present', function () {
 
 	afterEach(function () {
 		fixture.innerHTML = '';
-		axe._tree = undefined;
+		axe._tree = undefined;
```
Comment (marcysutton) [inline test/checks/navigation/internal-link-present.js]: Sidenote: should we move this `axe._tree` thing to the `checkContext.reset()` method?

File: test/checks/navigation/internal-link-present.js [hunk 2]
```diff
@@ -18,6 +18,16 @@ describe('internal-link-present', function () {
 		assert.isTrue(checks['internal-link-present'].evaluate.apply(checkContext, params));
 	});
 
+	it('should return false when a hashbang URL was used', function () {
+		var params = checkSetup('<div id="target"><a href="#!foo">hi</a></div>');
+		assert.isFalse(checks['internal-link-present'].evaluate.apply(checkContext, params));
+	});
+
+	it('should return false when a hash route URL was used', function () {
+		var params = checkSetup('<div id="target"><a href="#/home">hi</a></div>');
```
Comment (marcysutton) [inline test/checks/navigation/internal-link-present.js]: I'd like to see one more test case to mimic [OP's issue](https://twitter.com/ackleyk/status/981549994711896064): `href="#!/contact-us"`

File: test/checks/navigation/internal-link-present.js (hunk 1 above)
Comment (WilcoFiers) [reply test/checks/navigation/internal-link-present.js]: I think we're better off not storing the flat tree as a global.

**Pull request merged**